domain/application/service/application.go

```diff
@@ -200,6 +200,16 @@ type ApplicationState interface {
 	// be used as a cheap way to see if a charm exists without needing to load
 	// the charm metadata.
 	GetCharmIDByApplicationName(context.Context, string) (corecharm.ID, error)
+
+	// GetApplicationIDByUnitName returns the application ID for the named unit,
+	// returning an error satisfying [applicationerrors.UnitNotFound] if the unit
+	// doesn't exist.
+	GetApplicationIDByUnitName(ctx context.Context, name coreunit.Name) (coreapplication.ID, error)
```

Comment:
Seems useful, though in what situations will we have the unit name but not the unit UUID?

Reply:
In the facades where Entity is used over the wire, we only have the unit name from the tag.